Change power efficiency numbers

ymber opened the pull request:
#### Summary
```SUMMARY: Balance "Change power draw of vehicle parts and electric tools to reflect reality"```

#### Purpose of change
A lot of power draw numbers were a long way off from reality and/or hadn't been updated when everything switched to watts. With the recent and future changes increasing energy scarcity incredibly inefficient devices would become an obvious problem. Imminent changes to make turn and action lengths more realistic will also make such inefficient hardware look more like 0A era kind of artistic license physics.

#### Describe the solution
Change values for electric power draw to accurately reflect reality. In most cases this has meant significantly increasing energy efficiency.

#### Additional context
The vehicle epower changes should be fine now but changes to turns_per_charge values shouldn't be merged until after #30253 or electric tools will be six times more efficient than they should be.

mlangsdorf commented:
Please update the docs so future contributors know how to convert watts to turns_per_charge.

Also consider replacing turn_per_charge with "watts" as a float in JSON and then converting it in the code.  That would improve readability and make it much easier to convert to 1 second turns in the future.

anothersimulacrum commented on data/json/items/tool_armor.json:
```diff
@@ -980,9 +980,10 @@
+    "description": "A pair of battery-powered light amplifying goggles with an infrared illuminator, allowing you to see in the dark.  It is turned on, and continually draining batteries.  Use it to turn them off.",
     "flags": [ "GNV_EFFECT", "FRAGILE", "TRADER_AVOID" ],
-    "turns_per_charge": 9,
+    "//": "2019 commercial models can operate at under 0.375W with the IR illuminator on"
```
```suggestion
    "//": "2019 commercial models can operate at under 0.375W with the IR illuminator on",
```

Inglonias commented:
Are you planning to adjust battery capacity in a similar fashion? Right now they use round numbers that don't make a whole lot of sense, and altering efficiency means that altering battery capacity might also be called for.

Inglonias commented on data/json/items/tools.json:
```diff
@@ -5209,7 +5209,7 @@
     "name_plural": "mp3 players (on)",
     "description": "This mp3 player is turned on and playing some great tunes, raising your morale steadily while on your person.  It runs through batteries quickly; you can turn it off by using it.  It also obscures your hearing.",
-    "turns_per_charge": 150,
+    "turns_per_charge": 10000,
```
This is a ludicrous increase in efficiency. I wonder if we'll see any bug reports about tools suddenly having infinite battery life?

ymber commented:
Added a new power_draw field and updated most items to use it. We should consider using joules internally for storing energy. Working accurately with integer kJ values is hard with low power over short times.

anothersimulacrum commented on src/map.cpp:
```diff
@@ -4559,13 +4559,12 @@ static void process_vehicle_items( vehicle &cur_veh, int part )
             }
             if( n.ammo_capacity() > n.ammo_remaining() ) {
-                constexpr int per_charge = 10;
+                // Around 85% efficient, so double discharge every once every 7 seconds
```
```suggestion
                // Around 85% efficient, so double discharge once every 7 seconds
```
